spec/factories/subjects/primary_subjects.rb

```diff
@@ -0,0 +1,63 @@
+FactoryBot.define do
+  subjects = [
+     { subject_name: "Primary", subject_code: "00" },
+     { subject_name: "Primary with English", subject_code: "01" },
+     { subject_name: "Primary with geography and history", subject_code: "02" },
+     { subject_name: "Primary with mathematics", subject_code: "03" },
+     { subject_name: "Primary with modern languages", subject_code: "04" },
+     { subject_name: "Primary with physical education", subject_code: "06" },
+     { subject_name: "Primary with science", subject_code: "07" },
+   ]
+
+  factory :primary_subject do
+    transient do
+      sample_subject { subjects.sample }
+    end
+
+    subject_name { sample_subject[:subject_name] }
+    subject_code { sample_subject[:subject_code] }
+    type { :PrimarySubject }
```

Comment:
Pretty sure we don't need this, and all the `type` settings below. Rails will set this when instantiating a `PrimarySubject` using the factory name.